File: Seunghwan/week7/BOJ_1920.py
n = int(input())
a = list(map(int,input().split()))
m = int(input())
b = list(map(int,input().split()))

a.sort()

def Binary(arr, val, low, high):
    if low > high:
        return False
    
    mid = (low + high) // 2
    if arr[mid] > val:
        return Binary(arr, val, low, mid - 1)
    elif arr[mid] < val:
        return Binary(arr, val, mid + 1, high)
    else:
        return True


for m in b:
    if Binary(a, m, 0, n-1):
        print(1)
    else:
        print(0)

File: KyeongYoon/Week7/BOJ1920.c
#include <stdio.h>
#define SIZE 100000

int A[SIZE];
int key[SIZE];

void bubbleSort(int arr[]);
int BSearch(int arr[], int, int);

int main(void) {
    int num1, num2;

    scanf("%d", &num1);
    for (int i = 0; i < num1; i++) {
        int ins;
        scanf("%d", &ins);
        A[i] = ins;
    }

    scanf("%d", &num2);
    for (int i = 0; i < num2; i++) {
        int ins;
        scanf("%d", &ins);
        key[i] = ins;
    }

    bubbleSort(A);

    for (int i = 0; i < num2; i++) {
        printf("%d\n", BSearch(A, key[i], num1));
    }
}

void bubbleSort(int arr[]) {
    for (int i = 0; i < sizeof(arr); i++) {
        for (int j = 0; j < sizeof(arr) - i; j++) {
            if (arr[i] > arr[j + 1]) {
                int temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
            }
        }
    }
}

int BSearch(int arr[], int target, int size) {
    int low = 0;
    int high = size - 1;
    int mid = 0;

    while(low <= high) {
        mid = (low + high) / 2;

        if (arr[mid] == target) return 1;
        if (arr[low] == target) return 1;
        if (arr[high] == target) return 1;
        else if (arr[mid] > target) high = mid - 1;
        else low = mid + 1;
    }
    return 0;
}